### PR TITLE
Stale warning

### PR DESCRIPTION
**Pull request by rahularya50**
Is this the kind of thing you want for #3?

### CODE REVIEW

**Comment by rahularya50** [comment]
Actually, thinking about it, should this kind of thing be added for the debugging + environment diagram (+ maybe even OKPy test cases?) views as well?

**Comment by kavigupta** [comment]
Yeah it should probably be added for all of those things.

**Comment by rahularya50** [comment]
So I've added that warning to the debugger + environment diagram. I'm leaving it off the OKPy tests for the time being, the rationale being that such a change should occur when addressing #8.

Does everything else look OK?

**Comment by kavigupta** [comment]
Could you somehow make it based not on whether the file has been touched but on whether the parsed file is different? Like run `scheme_read` and then hash it and check if that's different.

**Comment by rahularya50** [comment]
Sure, that sounds feasible.

On a separate note, what happens if the file has a (load ...) expression, and the file it refers to is modified? This is relevant for OKPy tests in particular.

**Comment by rahularya50** [comment]
> Could you somehow make it based not on whether the file has been touched but on whether the parsed file is different? Like run `scheme_read` and then hash it and check if that's different.

Done. I don't hash it, since string comparison is already O(n) so I didn't see the point in not storing the entire AST.

File: hw10.scm [hunk 1]
```diff
@@ -1,9 +1,9 @@
 (define (factorial x)
   (define (helper x curr)
-    (if (= x 1)
+    (if (= x 1)b
```
Comment (kavigupta) [inline hw10.scm:3]: Don't include modifications to test files in the commit.

**Review by kavigupta — APPROVED** [review]
I've requested one minor change, otherwise LGTM

**Pull request merged**